policies/trademark-policy.md

```diff
@@ -1,172 +1,325 @@
-  For a detailed list, visit [Commonhaus Trademarks](https://www.commonhaus.org/trademarks/).
+#### Distribution of unmodified source code or artifacts packaged by us
 
-- Use the full form of any trademarks (e.g., Commonhaus Foundation”) in the first reference in all documents of mass communication, including marketing collateral and web pages.
-  You may then use any abbreviated or short form references (e.g. CF) within the same document where the full form has already been used.
+When you redistribute an unmodified copy of our software, either in source code form or as packaged artifacts (such as executables, libraries, or jar files), you are not changing the quality or nature of the software.
 
-### Compatibility and Interoperability
+In this case, you may retain the Word Marks and the Logos we have placed on the software to identify your redistribution.
+This kind of use only applies if you are redistributing an official distribution from this Project that has not been changed in any way.
+
+You can find files for the Logos and approved designs for packaging on our [Brand Standards][] page.
+
+#### Distribution of modified source code or artifacts packaged you
+
+If you compile and distribute bitwise identical packaged artifacts from our unmodified source code, you may retain our Word Marks and Logos.
+
+You may use the Word Marks, but not the Logos, to truthfully describe the origin of the software that you are providing, that is, that the code you are distributing is a modification of our software. You may also say, for example, that "this software is derived from the source code for Hibernate version 5.2."
+
+If your modification includes minor changes like bug fixes that preserve compatibility with our published artifacts, your artifacts should be labeled to indicate they were built by you.
+For instance, the Maven dependency io.quarkus:quarkus-arc:3.8.5.redhat-00004 clearly indicates it was built by Red Hat.
+
+Substantive modifications, on the other hand, should be treated as a fork. For example, where the functionality diverges, group or artifact identifiers should be changed to ensure users are not confused about which variant of the software they are getting and from whom.
+You may place your own trademarks or logos on forked versions of the software, as you have become the origin of that version. In such cases, you should not use our Logos.
+
+#### Statements about compatibility, interoperability or derivation
 
 You may use the Word Marks, but not the Logos, to truthfully describe the relationship between your software and ours.
-The use of the Mark cannot be used to imply that the Commonhaus Foundation or its projects are sponsoring or endorsing your product.
 
-The following are examples of proper and improper usage of these trademarks of the CF Marks:
+Our Mark should be used after a verb or preposition that describes the relationship between your software and ours.
```

Comment:
i would like to continue to be able to say "OpenRewrite by Moderne" because we pour basically all the energy into it -- at least until that isn't true. this seems to prohibit that.

Reply:
I believe this would be specific permission requested by Moderne, which would fall under the "separate written permission" category.

As I'm thinking about it, this would be part of the contribution agreement between us (CF) and you (Moderne).